### PR TITLE
[TG-105][feat] Added separate processes for scenarios and scenarios stopping on remove

### PR DESCRIPTION
**Pull request by quasiunafantasia**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-backend/pull/15" target="_blank">ReviewNinja

File: src/scenarios/runner.js [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+        output: output_stream.next.bind(output_stream)
+    }
+};
 
-}
-
-export default { run };
\ No newline at end of file
+process.on('message', (message) => {
+    if (message.type === 'start') {
```
Comment (kosich) [inline src/scenarios/runner.js]: @quasiunafantasia , why do we need separate message for starting? can't we start right away?

File: src/scenarios/scenario.manager.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            content: message
+        });
+    });
+}
+
+function stop(scenario) {
+    let scenarioProcess = runningScenarios.get(scenario.id);
+    if (scenarioProcess) {
+        scenarioProcess.kill();
+        runningScenarios.delete(scenario.id);
```
Comment (kosich) [inline src/scenarios/scenario.manager.js]: scenario might have died on its own
like, due to some unhandled error
do we track such cases? if not -- lets **always** delete it from map in stop

File: src/scenarios/scenario.model.js [hunk 3]
```diff
@@ -55,7 +55,11 @@ ScenarioSchema.options.toJSON = ScenarioSchema.options.toObject = {
 };
 
 ScenarioSchema.post('save', function(doc) {
-    run(doc);
+    start(doc);
```
Comment (kosich) [inline src/scenarios/scenario.model.js]: rename it to `scenario` either then
looks like they are different entities

File: src/scenarios/runner.js (hunk 1 above)
Comment (quasiunafantasia) [inline src/scenarios/runner.js]: @kosich we have a common scenario runner, but different scenario bodies. i couldn`t find another way to make runner execute different scripts

**Comment by kosich** [comment]
⭐

**Pull request merged**